torch/fft/__init__.py

```diff
@@ -9,18 +9,283 @@
+    always be purely real.
+""")
+
+irfft = _add_docstr(_fft.fft_irfft, r"""
+irfft(input, n=None, dim=-1, norm=None) -> Tensor
+
+Computes the inverse of :func:`~torch.fft.rfft`.
+
+:attr:`input` is a one-sided Hermitian signal in the Fourier domain, as
+produced by :func:`~torch.fft.rfft`. By the Hermitian property, the time-domain
+signal will be purely real.
+
+Note:
+    Some input frequencies must be purely real to satisfy the Hermitian
+    property. In these cases the imaginary component will be ignored.
+
+Note:
+    The correct interpretation of the hermitian input depends on the length of
+    the original data, as given by :attr:`n`. This is because each input shape
+    could correspond to either an odd or even length signal. By default, the
+    signal is assumed to be even length and odd signals will not round-trip
+    properly. So, it is recommended to always pass the signal length :attr:`n`:
+
+Args:
+    input (Tensor): the input tensor representing a half-Hermitian signal
+    n (int, optional): Output signal length. This determines the length of the
+        output signal. If given, the input will either be zero-padded or trimmed to this
+        length before taking the real IFFT.
+        Defaults to even output: ``n=2*(input.size(dim) - 1)``.
+    dim (int, optional): The dimension along which to take the one dimensional real IFFT.
+    norm (str, optional): Normalization mode. Between the forward and backward
+        transforms (:func:`~torch.fft.rfft` and :func:`~torch.fft.irfft`) a total
+        normalization factor of ``1/n`` is applied. The mode determines where
+        this normalization is applied:
+
+        * ``"forward"`` - the forward transform is normalized by ``1/n``, backward is unnormalized.
+        * ``"backward"`` - the backward transform is normalized by ``1/n``, forward is unnormalized.
+        * ``"ortho"`` - both transforms are normalized by ``1/sqrt(n)`` (making it orthonormal).
+
+        Default is ``"backward"``.
+
+Example:
+
+    >>> import torch.fft
+    >>> t = torch.arange(5)
+    >>> t
+    tensor([0, 1, 2, 3, 4])
+    >>> T = torch.fft.rfft(t)
+    >>> T
+    tensor([10.0000+0.0000j, -2.5000+3.4410j, -2.5000+0.8123j])
+
+    Without specifying the output length to :func:`~torch.fft.irfft`, the output
+    will not round-trip properly because the input is odd-length:
+
+    >>> torch.fft.irfft(T)
+    tensor([0.6250, 1.4045, 3.1250, 4.8455])
+
+    So, it is recommended to always pass the signal length :attr:`n`:
+
+    >>> torch.fft.irfft(T, t.numel())
+    tensor([0.0000, 1.0000, 2.0000, 3.0000, 4.0000])
+""")
+
+hfft = _add_docstr(_fft.fft_hfft, r"""
+hfft(input, n=None, dim=-1, norm=None) -> Tensor
+
+Computes the one dimensional discrete Fourier transform of a Hermitian
+symmetric :attr:`input` signal.
+
+Because the signal is Hermitian in the time-domain, the result will be real in
+the frequency domain. Note that some input frequencies must be purely real to
```

Comment:
Note should probably be a "Note:".

I have the same question here as above about ignoring the imaginary component. One way to deal with this might be to include an example, too.